services/identity/src/EShopOnAbp.IdentityService.Application/UserLoggedInEventHandler.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Threading.Tasks;
+using EShopOnAbp.IdentityService.ETOs;
+using Microsoft.Extensions.Logging;
+using Volo.Abp;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Distributed;
+using Volo.Abp.Identity;
+using Volo.Abp.Uow;
+
+namespace EShopOnAbp.IdentityService;
+
+public class UserLoggedInEventHandler : IDistributedEventHandler<UserLoggedInEto>, ITransientDependency
+{
+    private readonly IdentityUserManager _userManager;
+    private readonly ILogger<UserLoggedInEventHandler> _logger;
+
+    public UserLoggedInEventHandler(IdentityUserManager userManager, ILogger<UserLoggedInEventHandler> logger)
+    {
+        _userManager = userManager;
+        _logger = logger;
+    }
+
+
+    [UnitOfWork]
+    public async virtual Task HandleEventAsync(UserLoggedInEto eventData)
+    {
+        if (eventData == null)
+        {
+            _logger.LogWarning($"Handling UserLoggedInEvent failed! No user information found!");
+            return;
+        }
+
+        var user = await _userManager.FindByIdAsync(eventData.Id.ToString());
+
+        if (user == null)
+        {
+            await CreateCurrentUserAsync(eventData);
+        }
+        else
+        {
+            await UpdateCurrentUserAsync(user, eventData);
+        }
+    }
+
+    protected virtual async Task CreateCurrentUserAsync(UserLoggedInEto userInfo)
+    {
+        var user = new IdentityUser(
+            userInfo.Id,
+            userInfo.UserName,
+            userInfo.Email);
+
+        user.SetEmailConfirmed(userInfo.IsEmailVerified);
+
+        if (!string.IsNullOrEmpty(userInfo.Phone))
+        {
+            user.SetPhoneNumber(userInfo.Phone, false);
+        }
+
+        // This should run once to sync the admin userIds that seeded by IdentityModule and the Keycloak admin
```

Comment:
How about updating DataSeeding instead of deleting already seeded data?

Reply:
Keycloak needs an admin user so that we can use this user to get access token to use the keycloak API. 

It is set when we run the keycloak in the first place: https://github.com/abpframework/eShopOnAbp/blob/784dc99f60557611bb8609a8b6b0c82dd1d60d16/etc/docker/docker-compose.infrastructure.override.yml#L40-L41
We can only provide password for the admin user for keycloak admin user.

When the applications start running, an admin user is seeded by IdentityService. After this seeding, the IdentityService database admin user and the keycloak database admin user have **different Ids**.

To sync the initial admin users, I decided to remove the first seeded seeded admin user by IdentityService and re-create with the Id of the keycloak admin user. Admin user sync should only happen once.